api/client.go

```diff
@@ -237,3 +243,22 @@ func generateScopesSuggestion(statusCode int, endpointNeedsScopes, tokenHasScope
 
 	return ""
 }
+
+func clientOptions(hostname string, transport http.RoundTripper) ghAPI.ClientOptions {
+	// AuthToken, and Headers are being handled by transport,
+	// so let go-gh know that it does not need to resolve them.
+	opts := ghAPI.ClientOptions{
+		AuthToken: "none",
+		// Blank values for Accept, Authorization, Content-Type, Time-Zone, and User-Agent headers.
+		Headers: map[string]string{
+			accept:        "",
+			authorization: "",
+			contentType:   "",
+			timeZone:      "",
+			userAgent:     "",
```

Comment:
The semantics for passing a blank string to get the desired (non-blank) result in the end is confusing to me; would there be a way to make this clearer as follow-up?